tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/Execution/ProcessManager.cs

```diff
@@ -345,5 +365,31 @@ static List<int> GetChildrenPS (ILog log, int pid)
 
 			return list;
 		}
+
+		static string DetectXcodePath ()
+		{
+			using var process = new Process ();
+			process.StartInfo.FileName = "xcode-select";
+			process.StartInfo.Arguments = "-p";
+
+			var log = new MemoryLog ();
+			var stdout = new MemoryLog () { Timestamp = false };
+			var stderr = new ConsoleLog ();
+			var timeout = TimeSpan.FromSeconds (30);
+
+			var result = RunAsyncInternal (process, log, stdout, stderr, timeout).GetAwaiter ().GetResult ();
+
+			if (!result.Succeeded)
+				throw new Exception ("Failed to detect Xcode path from xcode-select!");
+
+			// Something like /Applications/Xcode114.app/Contents/Developers
+			var xcodeRoot = stdout.ToString ().Trim ();
+
+			if (string.IsNullOrEmpty (xcodeRoot))
+				throw new Exception ("Failed to detect Xcode path from xcode-select!");
+
+			// We need /Applications/Xcode114.app only
+			return Path.GetDirectoryName(Path.GetDirectoryName(xcodeRoot));
```

Comment:
minor: missing spaces before the `(`

Reply:
True but this code will be moved and reformatted next week so I would let it go in this time